pgmpy/factors/continuous/ContinuousFactor.py

```diff
@@ -0,0 +1,187 @@
+import six
+
+
+class ContinuousFactor(object):
+    """
+    Base class for factors representing various multivariate
+    representations.
+    """
+    def __init__(self, variables, pdf):
+        """
+        Parameters
+        ----------
+        variables: list or array-like
+            The variables for wich the distribution is defined.
+
+        pdf: function
+            The probability density function of the distribution.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from scipy.special import beta
+        # Two variable drichlet ditribution with alpha = (1,2)
+        >>> def drichlet_pdf(x, y):
+        ...     return (np.power(x, 1)*np.power(y, 2))/beta(x, y)
+        >>> from pgmpy.factors import ContinuousFactor
+        >>> drichlet_factor = ContinuousFactor(['x', 'y'], drichlet_pdf)
+        >>> drichlet_factor.scope()
+        ['x', 'y']
+        >>> drichlet_factor.assignemnt(5,6)
+        226800.0
+        """
+        if isinstance(variables, six.string_types):
+            raise TypeError("Variables: Expected type list or array like, got string")
```

Comment:
@yashu-seth Wouldn't checking if it's array type better ? And then you can throw an error that `expected type list or array, got ...`